subprojects/diagnostics/src/integTest/groovy/org/gradle/api/tasks/diagnostics/OutgoingVariantsReportTaskIntegrationTest.groovy

```diff
@@ -108,6 +108,112 @@ Secondary variants (*)
         hasSecondaryVariantsLegend()
     }
 
+    def "reports outgoing variants of a Java Library with documentation"() {
+        buildFile << """
+            plugins { id 'java-library' }
+            java {
+                publishJavadoc()
+                publishSources()
+            }
+            group = 'org'
+            version = '1.0'
+        """
+
+        when:
+        run ':outgoingVariants'
+
+        then:
+        outputContains """> Task :outgoingVariants
+--------------------------------------------------
+Variant apiElements
+--------------------------------------------------
+Description = API elements for main.
+
+Capabilities
+    - org:myLib:1.0 (default capability)
+Attributes
+    - org.gradle.category            = library
+    - org.gradle.dependency.bundling = external
+    - org.gradle.jvm.version         = ${JavaVersion.current().majorVersion}
+    - org.gradle.libraryelements     = jar
+    - org.gradle.usage               = java-api
+
+Artifacts
+    - build${File.separator}libs${File.separator}myLib-1.0.jar (artifactType = jar)
+
+Secondary variants (*)
+    - Variant : classes
+       - Attributes
+          - org.gradle.category            = library
+          - org.gradle.dependency.bundling = external
+          - org.gradle.jvm.version         = ${JavaVersion.current().majorVersion}
+          - org.gradle.libraryelements     = classes
+          - org.gradle.usage               = java-api
+       - Artifacts
+          - build${File.separator}classes${File.separator}java${File.separator}main (artifactType = java-classes-directory)
+    - Variant : javadoc
```

Comment:
I wonder about this choice. The `java-runtime` vs `java-api` aspect of the Javadoc kind of depends on the way the `javadoc` task is configured.
Would it not be better to simply have the default `source` and `javadoc` on the `java-runtime` bits? Knowing that users can be more specific if needed.